codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -95,22 +119,31 @@ protected boolean isAggregated(HttpObject msg) throws Exception {
     }
 
     @Override
-    protected boolean hasContentLength(HttpMessage start) throws Exception {
-        return HttpHeaderUtil.isContentLengthSet(start);
+    protected boolean isContentLengthInvalid(HttpMessage start, int maxContentLength) {
+        return getContentLength(start, -1) > maxContentLength;
     }
 
     @Override
-    protected long contentLength(HttpMessage start) throws Exception {
-        return HttpHeaderUtil.getContentLength(start);
+    protected Object newContinueResponse(HttpMessage start, int maxContentLength, ChannelPipeline pipeline) {
+        if (HttpHeaderUtil.is100ContinueExpected(start)) {
+            if (getContentLength(start, -1) <= maxContentLength) {
+                return CONTINUE.duplicate().retain();
+            }
+
+            pipeline.fireUserEventTriggered(HttpExpectationFailedEvent.INSTANCE);
+            return EXPECTATION_FAILED.duplicate().retain();
+        }
+        return null;
     }
 
     @Override
-    protected Object newContinueResponse(HttpMessage start) throws Exception {
-        if (HttpHeaderUtil.is100ContinueExpected(start)) {
-            return CONTINUE;
-        } else {
-            return null;
-        }
+    protected boolean closeAfterContinueResponse(Object msg) {
+        return closeOnExpectationFailed && ignoreContentAfterContinueResponse(msg);
+    }
+
+    @Override
+    protected boolean ignoreContentAfterContinueResponse(Object msg) {
+        return msg instanceof HttpResponse && ((HttpResponse) msg).status() == HttpResponseStatus.EXPECTATION_FAILED;
```

Comment:
@Scottmitch  I think it is better to compare the `code()` as a user could construct a HttpResponseStatus.